### PR TITLE
isEqual in models

### PR DESCRIPTION
**Pull request by jaelim-ms**

### CODE REVIEW

File: MobileCenter/MobileCenter/Internals/Model/MSLogContainer.h [hunk 1]
```diff
@@ -14,30 +14,36 @@
...
- * logs Array of logs
+ * @return A log container instance for the given batch ID.
  */
 - (id)initWithBatchId:(NSString *)batchId andLogs:(NSArray<MSLog> *)logs;
 
 /**
- * Serialize logs into a JSON string
+ * Serialize logs into a JSON string.
+ *
+ * @return A JSON stringl.
```
Comment (erychagov) [inline MobileCenter/MobileCenter/Internals/Model/MSLogContainer.h]: typo, _string_

File: MobileCenter/MobileCenter/MSWrapperSdk.h [hunk 2]
```diff
@@ -49,9 +40,8 @@
 /**
  * Checks if the object's values are valid.
  *
- * return YES, if the object is valid
+ * @return YES, if the object is valid
```
Comment (erychagov) [inline MobileCenter/MobileCenter/MSWrapperSdk.h]: dot?

File: MobileCenter/MobileCenter/Internals/Model/MSLogContainer.h (hunk 1 above)
Comment (jaelim-ms) [inline MobileCenter/MobileCenter/Internals/Model/MSLogContainer.h]: 😮 Fixed.

File: MobileCenter/MobileCenter/MSWrapperSdk.h (hunk 2 above)
Comment (jaelim-ms) [inline MobileCenter/MobileCenter/MSWrapperSdk.h]: 😮 Fixed.

File: MobileCenter/MobileCenter/MSWrapperSdk.h [hunk 3]
```diff
@@ -31,15 +31,6 @@
...
-/**
- * Is equal to another wrapper SDK
- *
- * @param object Wrapper SDK
- *
- * @return Return YES if equal and NO if not equal
- */
-- (BOOL)isEqual:(id)object;
-
 - (instancetype)initWithWrapperSdkVersion:(NSString *)wrapperSdkVersion
```
Comment (TroubleMakerBen) [inline MobileCenter/MobileCenter/MSWrapperSdk.h]: Why no doc on this one?

**Comment by dhei** [comment]
Looks good to me.

File: MobileCenter/MobileCenter/MSWrapperSdk.h (hunk 3 above)
Comment (jaelim-ms) [inline MobileCenter/MobileCenter/MSWrapperSdk.h]: We have many methods without docs. We have to address that, can be a separate task.

File: MobileCenter/MobileCenter/Internals/Model/MSAbstractLog.m [hunk 4]
```diff
@@ -41,6 +41,17 @@ - (BOOL)isValid {
   return self.type && self.toffset && self.device && [self.device isValid];
 }
 
+- (BOOL)isEqual:(id)object {
+  if (![object isKindOfClass:[MSAbstractLog class]]) {
```
Comment (clpolet) [inline MobileCenter/MobileCenter/Internals/Model/MSAbstractLog.m]: You should also check for nullity here and return `NO` if that's the case.

File: MobileCenter/MobileCenter/MSWrapperSdk.h (hunk 3 above)
Comment (TroubleMakerBen) [inline MobileCenter/MobileCenter/MSWrapperSdk.h]: ok!

File: MobileCenter/MobileCenter/Internals/Model/MSLogWithProperties.m [hunk 5]
```diff
@@ -15,6 +15,14 @@ - (NSMutableDictionary *)serializeToDictionary {
   return dict;
 }
 
+- (BOOL)isEqual:(id)object {
+  if (![object isKindOfClass:[MSLogWithProperties class]] || ![super isEqual:object]) {
```
Comment (clpolet) [inline MobileCenter/MobileCenter/Internals/Model/MSLogWithProperties.m]: Check nil.

File: MobileCenter/MobileCenter/Internals/Model/MSStartServiceLog.m [hunk 6]
```diff
@@ -9,18 +9,25 @@ @implementation MSStartServiceLog
...
-  self = [super init];
-  if( self ) {
+  if ((self = [super init])) {
     self.type = kMSStartService;
   }
   return self;
 }
 
+- (BOOL)isEqual:(id)object {
+  if (![object isKindOfClass:[MSStartServiceLog class]] || ![super isEqual:object]) {
```
Comment (clpolet) [inline MobileCenter/MobileCenter/Internals/Model/MSStartServiceLog.m]: Check nil

File: MobileCenterAnalytics/MobileCenterAnalytics/Internals/Model/MSPageLog.m [hunk 7]
```diff
@@ -28,6 +28,14 @@ - (BOOL)isValid {
   return [super isValid] && self.name;
 }
 
+- (BOOL)isEqual:(id)object {
+  if (![object isKindOfClass:[MSPageLog class]] || ![super isEqual:object]) {
```
Comment (clpolet) [inline MobileCenterAnalytics/MobileCenterAnalytics/Internals/Model/MSPageLog.m]: Check nil.

File: MobileCenterCrashes/MobileCenterCrashes/Internals/Model/MSAbstractErrorLog.m [hunk 8]
```diff
@@ -71,7 +71,7 @@ - (BOOL)isValid {
 }
 
 - (BOOL)isEqual:(id)object {
-  if (!object || ![object isKindOfClass:[MSAbstractErrorLog class]]) {
```
Comment (clpolet) [inline MobileCenterCrashes/MobileCenterCrashes/Internals/Model/MSAbstractErrorLog.m]: Why remove the nil check here?

File: MobileCenterCrashes/MobileCenterCrashes/Internals/Model/MSAppleErrorLog.m [hunk 9]
```diff
@@ -85,7 +85,7 @@ - (BOOL)isValid {
 }
 
 - (BOOL)isEqual:(id)object {
-  if (!object || ![super isEqual:object] || ![object isKindOfClass:[MSAppleErrorLog class]]) {
+  if (![object isKindOfClass:[MSAppleErrorLog class]] || ![super isEqual:object]) {
```
Comment (clpolet) [inline MobileCenterCrashes/MobileCenterCrashes/Internals/Model/MSAppleErrorLog.m]: nil check?

File: MobileCenterCrashes/MobileCenterCrashes/Internals/Model/MSException.m [hunk 10]
```diff
@@ -49,7 +49,7 @@ - (BOOL)isValid {
 }
 
 - (BOOL)isEqual:(id)object {
-  if (!object || ![object isKindOfClass:[MSException class]]) {
+  if (![object isKindOfClass:[MSException class]]) {
```
Comment (clpolet) [inline MobileCenterCrashes/MobileCenterCrashes/Internals/Model/MSException.m]: nil check?

File: MobileCenterCrashes/MobileCenterCrashes/Internals/Model/MSStackFrame.m [hunk 11]
```diff
@@ -34,7 +34,7 @@ - (NSMutableDictionary *)serializeToDictionary {
 }
 
 - (BOOL)isEqual:(id)object {
-  if (!object || ![object isKindOfClass:[MSStackFrame class]]) {
+  if (![object isKindOfClass:[MSStackFrame class]]) {
```
Comment (clpolet) [inline MobileCenterCrashes/MobileCenterCrashes/Internals/Model/MSStackFrame.m]: Nil check?

File: MobileCenterCrashes/MobileCenterCrashes/Internals/Model/MSThread.m [hunk 12]
```diff
@@ -47,7 +47,7 @@ - (BOOL)isValid {
 }
 
 - (BOOL)isEqual:(id)object {
-  if (!object || ![object isKindOfClass:[MSThread class]]) {
+  if (![object isKindOfClass:[MSThread class]]) {
```
Comment (clpolet) [inline MobileCenterCrashes/MobileCenterCrashes/Internals/Model/MSThread.m]: Nil check?

File: MobileCenter/MobileCenter/Internals/Model/MSAbstractLog.m (hunk 4 above)
Comment (jaelim-ms) [inline MobileCenter/MobileCenter/Internals/Model/MSAbstractLog.m]: As mentioned offline, `isKindOfClass` will take care of it.

File: MobileCenter/MobileCenter/Internals/Model/MSLogWithProperties.m (hunk 5 above)
Comment (jaelim-ms) [inline MobileCenter/MobileCenter/Internals/Model/MSLogWithProperties.m]: As mentioned offline, `isKindOfClass` will take care of it.

File: MobileCenter/MobileCenter/Internals/Model/MSStartServiceLog.m (hunk 6 above)
Comment (jaelim-ms) [inline MobileCenter/MobileCenter/Internals/Model/MSStartServiceLog.m]: As mentioned offline, `isKindOfClass` will take care of it.

File: MobileCenterAnalytics/MobileCenterAnalytics/Internals/Model/MSPageLog.m (hunk 7 above)
Comment (jaelim-ms) [inline MobileCenterAnalytics/MobileCenterAnalytics/Internals/Model/MSPageLog.m]: As mentioned offline, `isKindOfClass` will take care of it.

File: MobileCenterCrashes/MobileCenterCrashes/Internals/Model/MSAbstractErrorLog.m (hunk 8 above)
Comment (jaelim-ms) [inline MobileCenterCrashes/MobileCenterCrashes/Internals/Model/MSAbstractErrorLog.m]: As mentioned offline, `isKindOfClass` will take care of it.

File: MobileCenterCrashes/MobileCenterCrashes/Internals/Model/MSAppleErrorLog.m (hunk 9 above)
Comment (jaelim-ms) [inline MobileCenterCrashes/MobileCenterCrashes/Internals/Model/MSAppleErrorLog.m]: As mentioned offline, `isKindOfClass` will take care of it.

File: MobileCenterCrashes/MobileCenterCrashes/Internals/Model/MSException.m (hunk 10 above)
Comment (jaelim-ms) [inline MobileCenterCrashes/MobileCenterCrashes/Internals/Model/MSException.m]: As mentioned offline, `isKindOfClass` will take care of it.

File: MobileCenterCrashes/MobileCenterCrashes/Internals/Model/MSStackFrame.m (hunk 11 above)
Comment (jaelim-ms) [inline MobileCenterCrashes/MobileCenterCrashes/Internals/Model/MSStackFrame.m]: As mentioned offline, `isKindOfClass` will take care of it.

File: MobileCenterCrashes/MobileCenterCrashes/Internals/Model/MSThread.m (hunk 12 above)
Comment (jaelim-ms) [inline MobileCenterCrashes/MobileCenterCrashes/Internals/Model/MSThread.m]: As mentioned offline, `isKindOfClass` will take care of it.

**Pull request merged**